wire-ios/Wire-iOS/Sources/Components/Settings/SettingsPropertyFactory.swift

```diff
@@ -231,12 +235,27 @@ final class SettingsPropertyFactory {
                 if var tracking = self.tracking {
                     switch value {
                     case .number(let number):
-                        tracking.disableAnalyticsSharing = number.boolValue
+                        let newValue = number.boolValue
+
+                        // If newValue is false, it means the user wants to enable analytics sharing
+                        if newValue == false {
+                            self.showAnalyticsConsentAlert { accepted in
+                                if accepted {
+                                    tracking.disableAnalyticsSharing = newValue
+                                    AVSFlowManager.getInstance()?.setEnableMetrics(!newValue)
```

Comment:
Isn't this part of the tracking manager? Why do it here as well?

Reply:
Moved it to the trackingManager: https://github.com/wireapp/wire-ios/commit/508f597ce80e6ab2a0a16c8dfb8b03f4c045901c